apix-impl/src/main/java/eu/xenit/apix/alfresco/dictionary/PropertyTypeCheckService.java

```diff
@@ -0,0 +1,68 @@
+package eu.xenit.apix.alfresco.dictionary;
+
+import eu.xenit.apix.data.QName;
+import eu.xenit.apix.properties.PropertyDefinition;
+import eu.xenit.apix.search.nodes.PropertySearchNode;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.function.Function;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class PropertyTypeCheckService {
+
+    private final static Logger logger = LoggerFactory.getLogger(PropertyTypeCheckService.class);
+
+    private final Map<String, Function<String, Boolean>> constraintMap = new HashMap<String, Function<String, Boolean>>() {{
+        put("d:int", PropertyTypeCheckService::isInt);
+        put("{http://www.alfresco.org/model/dictionary/1.0}int", PropertyTypeCheckService::isInt);
+        put("d:long", PropertyTypeCheckService::isLong);
+        put("{http://www.alfresco.org/model/dictionary/1.0}long", PropertyTypeCheckService::isLong);
+    }};
+
+    private final PropertyService propertyService;
+
+    public PropertyTypeCheckService(PropertyService propertyService) {
+        if (propertyService == null) {
+            logger.warn("Instantiating PropertyTypeCheckService without a PropertyService, type checks will always succeed");
+        }
+        this.propertyService = propertyService;
+    }
+
+    public boolean fitsType(PropertySearchNode node) {
+        if (node == null || node.getValue() == null || node.getName() == null || propertyService == null) {
+            return true;
+        }
+        QName qName = new QName(node.getName().replaceAll("\\\\", "")); //stuff like hyphen in node-dbid is escaped
```

Comment:
isn't there an API-way to do this ? feels like we're missing potential edge cases here ?

Reply:
Good question, I will look into it

Reply:
Didn't find anything with some quick searches
I abstracted out the existing escape mechanism and added an inverse (and some tests on them) as a proposal
https://github.com/xenit-eu/alfred-api/pull/44/commits/2aac5d36004c0c3b1c02f51e9a571f451931d321

Reply:
See comment on that commit.